src/sample/java/com/strangeberry/jmdns/tools/Browser.java

```diff
@@ -108,157 +111,149 @@ public class Browser extends JFrame implements ServiceListener, ServiceTypeListe
         info.setBackground(bg);
         info.setEditable(false);
         info.setLineWrap(true);
+        info.setFont(new Font(Font.MONOSPACED, Font.PLAIN, 12)); // Set fixed-width font
 
         JPanel infoPanel = new JPanel();
         infoPanel.setLayout(new BorderLayout());
         infoPanel.add("North", new JLabel("Details"));
         infoPanel.add("Center", new JScrollPane(info, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED));
         content.add(infoPanel);
-
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        setLocation(100, 100);
-        setSize(600, 400);
-
-        this.jmmdns.addServiceTypeListener(this);
-
-        // register some well known types
-        // String list[] = new String[] { "_http._tcp.local.", "_ftp._tcp.local.", "_tftp._tcp.local.", "_ssh._tcp.local.", "_smb._tcp.local.", "_printer._tcp.local.", "_airport._tcp.local.", "_afpovertcp._tcp.local.", "_ichat._tcp.local.",
-        // "_eppc._tcp.local.", "_presence._tcp.local.", "_rfb._tcp.local.", "_daap._tcp.local.", "_touchcs._tcp.local." };
-        String[] list = new String[] {};
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        int width = screenSize.width / 2; 
+        int height = screenSize.height / 2;
+        setSize(width, height);
+        setLocationRelativeTo(null);
 
-        for (int i = 0; i < list.length; i++) {
-            this.jmmdns.registerServiceType(list[i]);
-        }
+        Thread jmdnsLoaderThread = new Thread(new JmDnsLoader());
+        jmdnsLoaderThread.start();
 
         this.setVisible(true);
     }
 
-    /*
-     * (non-Javadoc)
-     * @see javax.jmdns.ServiceListener#serviceAdded(javax.jmdns.ServiceEvent)
+    /**
+     * Main program entry point.
+     *
+     * @param argv Command line arguments (not used)
+     */
+    public static void main(String[] argv) {
+        new Browser();
+    }
+
+    /**
+     * Called when a service is added.
+     *
+     * @param event The ServiceEvent containing information about the added service
      */
     @Override
     public void serviceAdded(ServiceEvent event) {
         final String name = event.getName();
-
-        System.out.println("ADD: " + name);

```

Comment:
Maybe it is just my personal preference to like reading logs but what I suggest is to create a private method where you do the logging. By default it is switched off and if someone who feels the irresistible need could switch it back on.